permission/src/main/java/org/wildfly/security/permission/LongNameSetPermissionCollection.java

```diff
@@ -103,7 +99,7 @@ public Permission nextElement() {
             }
             long bit = Long.lowestOneBit(bits);
             this.bits = bits & ~bit;
-            return permissionFor(Long.numberOfTrailingZeros(bit));
+            return ((AbstractNamedPermission<?>)getSourcePermission()).withName(getNameEnumeration().nameOf(Long.numberOfTrailingZeros(bit)));
```

Comment:
i think we an keep this as its own function for clarity, and just move the function into the `Iter`class

Reply:
changes done. kindly review